iphone/Classes/TiUIWindowProxy.m

```diff
@@ -937,6 +940,49 @@ - (void)cleanupWindowDecorations
     [barImageView removeFromSuperview];
   }
 }
+
+- (void)processForSafeArea
+{
+  if (self.shouldExtendSafeArea) {
+    return;
+  }
+#if IS_XCODE_9
+  float left = 0.0;
+  float right = 0.0;
+  float top = 0.0;
+  float bottom = 0.0;
+  if ([TiUtils isIOS11OrGreater]) {
+    UIViewController<TiControllerContainment> *topContainerController = [[[TiApp app] controller] topContainerController];
+    UIView *rootView = [topContainerController hostingView];
+    if (self.tabGroup) {
+      UIInterfaceOrientation orientation = [[UIApplication sharedApplication] statusBarOrientation];
+      if (!UIInterfaceOrientationIsPortrait(orientation)) {
+        left = rootView.safeAreaInsets.left;
+        right = rootView.safeAreaInsets.right;
+      }
+    } else if (self.tab) {
+      UIInterfaceOrientation orientation = [[UIApplication sharedApplication] statusBarOrientation];
+      if (!UIInterfaceOrientationIsPortrait(orientation)) {
+        left = rootView.safeAreaInsets.left;
+        right = rootView.safeAreaInsets.right;
+        bottom = rootView.safeAreaInsets.bottom;
+      } else {
+        bottom = rootView.safeAreaInsets.bottom;
+      }
+    } else {
+      left = rootView.safeAreaInsets.left;
+      right = rootView.safeAreaInsets.right;
+      bottom = rootView.safeAreaInsets.bottom;
+      top = rootView.safeAreaInsets.top;
+    }
+    [self.safeAreaViewProxy setTop:[NSNumber numberWithFloat:top]];
```

Comment:
- Use `NUMFLOAT` for all 4 sizes
- Access `safeAreaViewProxy` via `[self safeAreaViewProxy]`